### ISSUE TITLE
nn: random numbers

### ISSUE THREAD

**Issue by deasmhumhna** [gorgonia/gorgonia#259]
Any interest in refactoring the binomial and normal nodes to use `gonum/stat` inferfaces and allow a larger variety of/user defined distributions? Perhaps create a separate subpackage just for random numbers?

**Comment by chewxy** [comment]
YES!. This has been in the plans for ages. See also: github.com/gorgonia/randomkit.

I haven't had the time to do so tho.
